src/components/play/PlayUI.js

```diff
@@ -1581,21 +1610,711 @@ class PlayUI extends Component {
+                </svg>`)
+
+            const tempoButtonX = xOffset + HTML_UI_Params.tempoButtonXOffset
+            const tempoButtonY = yOffset + HTML_UI_Params.tempoButtonYOffset
+
+            const tempoIconX = xOffset + HTML_UI_Params.tempoIconXOffset
+            const tempoIconY = yOffset + HTML_UI_Params.tempoIconYOffset
+
+            const tempoButtonTextX = xOffset + HTML_UI_Params.tempoButtonTextXOffset
+            const tempoButtonTextY = yOffset + HTML_UI_Params.tempoButtonTextYOffset
+
+            tempoIcon.x(tempoIconX)
+            tempoIcon.y(tempoIconY)
+
+            tempoButton.x(tempoButtonX)
+            tempoButton.y(tempoButtonY)
+
+            tempoButton.fill('#fff').attr({ opacity: 0.1 })
+            const tempoButtonText = this.container.nested().plain(round.bpm)
+
+            tempoButtonText.x(tempoButtonTextX)
+            tempoButtonText.y(tempoButtonTextY)
+
+            tempoButtonText.font({
+                family: 'Arial',
+                size: 11,
+                weight: 900,
+                opacity: 1,
+            })
+            tempoButtonText.fill('#fff')
+
+            this.renderPlayingSequenceIndicator({ x: xOffset, y: yOffset })
+            this.renderRecordSequenceButton(xOffset, yOffset)
+        }
+    }
+
+    onLoadPattern = async (id) => {
+        if (!this.props.display.isRecordingSequence) {
+            const pattern = _.find(this.props.round.userPatterns[this.props.user.id].patterns, { id })
+            if (!_.isEmpty(pattern.state)) {
+                this.setState({ selectedPattern: pattern.id })
+                this.selectedPatternNeedsSaving = false
+
+                // check if we have layers in the round not referenced in the pattern then set all steps in that layer to off
+                for (const existingLayer of this.props.round.layers) {
+                    if (_.isNil(_.find(pattern.state.layers, { id: existingLayer.id })) && existingLayer.createdBy === this.props.user.id) {
+                        let existingLayerClone = _.cloneDeep(existingLayer)
+                        for (const step of existingLayerClone.steps) {
+                            step.isOn = false
+                        }
+                        pattern.state.layers.push(existingLayerClone)
+                    }
+                }
+
+                // check we haven't deleted the layer that is referenced in the pattern
+                let layersToDelete = []
+                for (const layer of pattern.state.layers) {
+                    const layerExists = _.find(this.props.round.layers, { id: layer.id })
+                    if (_.isNil(layerExists)) {
+                        layersToDelete.push(layer)
+                    }
+                }
+
+                _.remove(pattern.state.layers, function (n) {
+                    return layersToDelete.indexOf(n) > -1
+                })
+
+                await this.patternLayersToRound(pattern)
```

Comment:
just out of curiosity, why the await here?

Reply:
We need to wait for the writes to firebase to be complete, otherewise there's a bug that messes up the functionality esp on slower networks